### PR TITLE
check trie integrity in tests & fix more edge cases

### PR DESCRIPTION
**Pull request by dessaya**

### CODE REVIEW

File: clients/iota-go/iotaclient/iotaclienttest/api_governance_read_test.go [hunk 1]
```diff
@@ -18,7 +18,8 @@ func TestGetCommitteeInfo(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, epochId, committeeInfo.EpochId)
 	// just use a arbitrary big number to ensure there are enough validator
-	require.Greater(t, len(committeeInfo.Validators), 3)
+	// TODO: is this assertion needed? why not >= 1?
+	// require.Greater(t, len(committeeInfo.Validators), 3)
```
Comment (howjmay) [inline clients/iota-go/iotaclient/iotaclienttest/api_governance_read_test.go]: I ran this test on my local side without this change. The one before this change passed the test

Comment (dessaya) [reply clients/iota-go/iotaclient/iotaclienttest/api_governance_read_test.go]: did you do `docker pull iotaledger/iota-tools:devnet` ?

Comment (howjmay) [reply clients/iota-go/iotaclient/iotaclienttest/api_governance_read_test.go]: Based on the iota implementation [`crates/iota/src/iota_commands.rs:74`](https://github.com/iotaledger/iota/blob/develop/crates/iota/src/iota_commands.rs#L74), I think we should set this to 
`require.Equal(t, len(committeeInfo.Validators), 1)`

Comment (lmoe) [reply clients/iota-go/iotaclient/iotaclienttest/api_governance_read_test.go]: yup, seems like it :)

**Review by lmoe — APPROVED** [review]
